cmd/incus-migrate/main_migrate.go

```diff
@@ -348,6 +350,21 @@ func (c *cmdMigrate) RunInteractive(server incus.InstanceServer) (cmdMigrateData
 			return err
 		}
 
+		// When migrating a VM, ensure we only process supported inputs
+		if config.InstanceArgs.Type == api.InstanceTypeVM {
+			if linux.IsBlockdevPath(s) {
+				// pass
+			} else if _, ext, _, _ := archive.DetectCompression(s); ext == ".qcow2" {
+				// pass
+			} else if strings.HasSuffix(s, ".raw") {
+				// TODO -- need to actually inspect the image for MBR/GPT header
```

Comment:
Still trying to figure out what the best route would be to reliably identify raw images. There's github.com/diskfs/go-diskfs, but that might be a bit heavy just for this. Could also call `file` or `fdisk` to identify the image.

Reply:
I'd probably do it as a fallback, so if we don't see the magic for QCOW2 or VMDK, just assume it's a raw image.

We can have the tool indicate the detected format in the overview.